x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/loadingservice/ModelLoadingService.java

```diff
@@ -339,21 +375,23 @@ private void handleLoadSuccess(String modelId,
             trainedModelConfig.getDefaultFieldMap(),
             inferenceConfig,
             trainedModelConfig.getLicenseLevel(),
-            modelStatsService);
+            modelStatsService,
+            trainedModelCircuitBreaker);
         synchronized (loadingListeners) {
             listeners = loadingListeners.remove(modelId);
             // If there is no loadingListener that means the loading was canceled and the listener was already notified as such
             // Consequently, we should not store the retrieved model
             if (listeners == null) {
-                trainedModelCircuitBreaker.addWithoutBreaking(-inferenceDefinition.ramBytesUsed());
+                loadedModel.release();
                 return;
             }
+            for (ActionListener<LocalModel> listener = listeners.poll(); listener != null; listener = listeners.poll()) {
+                loadedModel.acquire();
+                listener.onResponse(loadedModel);
```

Comment:
I am not sure about calling `listener.onResponse(loadedModel);` while locked. 

We have no control over what `listener.onResponse` executes. 

Do you think it is OK?